src/core/dbus-unit.c

```diff
@@ -1674,9 +1674,17 @@ static int bus_unit_set_transient_property(
                         return r;
 
                 STRV_FOREACH(p, l) {
+                        path_simplify(*p, true);
+
                         if (!path_is_absolute(*p))
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Path specified in %s is not absolute: %s", name, *p);
 
+                        if (!path_is_normalized(*p))
```

Comment:
path_is_normalized() calls path_is_valid() as first thing, hence having both this check and the explicit path_is_valid() check in place in this order won't do any good.

Reply:
Also, that code duplication seems sub-optimal.  That also means those simplification and checks are performed twice, both here and in `unit_require_mounts_for`.

Reply:
@stephane-chazelas Theoretically, all input should be checked earlier. Otherwise, it may be hard to investigate what causes errors. Of course, if the early checks are perfect, your fix is not necessary, but for safety, it is better to also check in `unit_require_mounts_for()`.

Most (I hope all, but not sure) paths provided by unit files are already checked well, but arguments provided through DBus API may not. And actually your issue was caused by one of that. I will check other DBus API that later.

Reply:
> That also means those simplification and checks are performed twice, both here and in `unit_require_mounts_for`.

This is not an inner loop, we shouldn't optimize at that level. Yes, the check is done on multiple levels, but that's because we have multiple entry points to the inner level, and the outer level allows us to return more useful errors and uses the values for other stuff too